python/lsst/metaserv/metaREST_v0.py

```diff
@@ -24,134 +24,134 @@
-    return runDbQuery1(
-        "SELECT DDT_Table.* FROM DDT_Table JOIN DbMeta USING (dbMetaId) "
-        "WHERE dbName=%s AND tableName=%s",
-        (dbName, tableName),
-        "Table '%s.%s'not found." % (dbName, tableName))
+    query = "SELECT DDT_Table.* FROM DDT_Table " \
+            + "JOIN DbMeta USING (dbMetaId) " \
+            + "WHERE dbName=:dbName AND tableName=:tableName"
+    message = "No DDT_Table entries for database '%s'  and table %s." %(dbName, tableName)
+    return _resultOf(text(query), paramMap={"dbName":dbName, "tableName":tableName}, message=message)
 
 @metaREST.route('/db/<string:lsstLevel>/<string:dbName>/' +
                 'tables/<string:tableName>/schema', methods=['GET'])
 def getDbPerTypeDbNameTablesTableNameSchema(lsstLevel, dbName, tableName):
     '''Retrieves schema for a given table.'''
-    return runDbQuery1(
-        "SHOW CREATE TABLE %s.%s" % (dbName, tableName),
-        None,
-        "Table '%s.%s'not found." % (dbName, tableName))
+    # Scalar
+    if SAFE_SCHEMA_PATTERN.match(dbName) and SAFE_TABLE_PATTERN.match(tableName):
+        query = "SHOW CREATE TABLE %s.%s" % (dbName, tableName)
+        message = "Table '%s.%s'not found." % (dbName, tableName)
+        return _resultOf(query, message=message)
+    return _response(_error("ValueError", "Database name or Table name is not safe"))
 
 @metaREST.route('/image', methods=['GET'])
 def getImage():
     return ("meta/.../image not implemented. I am supposed to print list of "
             "supported image types here, something like: raw, template, coadd, "
             "jpeg, calexp, ... etc")
+
+_error = lambda exception, message: {"exception":exception, "message": message}
+_vector = lambda results: {"results": results}
+_scalar = lambda result: {"result": result}
+
+def _resultOf(query, paramMap=None, exception="NotFound",
+              message="No Entity Found"):
+    status_code = OK
+    paramMap = paramMap if paramMap else {}
+    try:
+        engine = current_app.config["default_engine"]
+        result = engine.execute(query, **paramMap).first()
+        if result:
+            response = _scalar([i for i in result])
+        else:
+            response = _error(exception, message)
+            status_code = NOT_FOUND
+    except SQLAlchemyError as e:
+        status_code = INTERNAL_SERVER_ERROR
+        response = _error(type(e).__name__, e.message)
+    return _response(response, status_code)
+
+def _resultsOf(query, paramMap=None, exception="NotFound",
+               message="No Entity Found"):
+    status_code = OK
+    log.info(query)
+    paramMap = paramMap if paramMap else {}
+    try:
+        engine = current_app.config["default_engine"]
+        results = [[i for i in result] for result in engine.execute(query, **paramMap)]
+        if len(results):
+            response = _vector(results)
+        else:
+            response = _error(exception, message)
+            status_code = NOT_FOUND
```

Comment:
In all cases in the above code, I communicate a message conveying that there are no objects found. In a few cases, those errors may be ambiguous ( as in getDbPerTypeDbNameTablesTableName), but we are aware of this and there's issue [DM-2537](https://jira.lsstcorp.org/browse/DM-2537) which specifically addresses this. 

I think DM-2537 should clear this up when I get to it.